### ISSUE TITLE
Calendar integration

### ISSUE THREAD

**Issue by nursoda** [TB-throwback/LookOut-fix-version#27]
# Prerequisites
- I am running the latest version
- I checked the documentation and found no answer
- I checked to make sure that this issue has not already been filed

## Context
* Operating System: Solus (Arch Linux)
* Thunderbird version number: 60.3.1
* 64 bit/32 bit: 64 bit
* LookOut (fix version) version number: 1.2.23

# Expected Behavior
If I receive a (TNEF encoded) Outlook/Exchange calender invitation, I expect to be able to accept or decline that meeting.

# Current Behavior
Currently, all I see is three attachments that I could save - and do what?

**Comment by dugite-code** [comment]
If you have the open calendaring data directly enabled you can double click the ics file in order to add it into your calendar as normal.
![image](https://user-images.githubusercontent.com/12742064/49260187-6950f900-f477-11e8-8f46-61a3f5419d9c.png)

I need to look into how to bring up the normal lightning calendar notification. I believe when I finally figure out the inline attachments this will also be covered.

**Comment by dugite-code** [comment]
@nursoda I sent you an email. I have been having an issue with google putting my server's emails to spam lately so please check your spam folder (and mark as not spam)

**Comment by nursoda** [comment]
Apart from "@nursoda I sent you an email..." I did not receive any other
direct message from you. I don't have a spam folder, so no way it hides
there ;-)

**Comment by dugite-code** [comment]
If it's a gmail account there is a spam folder it might just be hidden. At any rate I've added a support email to the [ATN Page](https://addons.thunderbird.net/addon/lookout-fix-version/). I'll do my best to investigate the issue but just keep in mind it's all free time based.

**Comment by dugite-code** [comment]
Alright this looks like a confirmation of #3. I had hoped this was just a one off bug. It looks like the parser isn't converting something correctly

![image](https://user-images.githubusercontent.com/12742064/49621876-4dfb6600-fa02-11e8-874a-c34ac809272a.png)

**Comment by dugite-code** [comment]
New finding: The UID is also not being processed correctly

**Comment by dugite-code** [comment]
[Beta Release v1.2.25b1](https://github.com/TB-throwback/LookOut-fix-version/releases/tag/v1.2.25b1) has multiple changes attempting to fix these issues. Unfortunately I couldn't figure out a file save bug so I had to merge the drag and drop branch. There shouldn't be any performance impact but no promises

If you can test them it would be appreciated.

**Comment by dugite-code** [comment]
# Notes

On first open I am occasionally seeing this:
```
TypeError: array[idx] is null[Learn More]  tnef.js:1290:11
	rm_quotes chrome://lookout/content/tnef.js:1290:11
	forEach self-hosted:261:13
	tnef_pack_get_name_addr chrome://lookout/content/tnef.js:1307:5
	tnef_pack_appt_attendees_ics chrome://lookout/content/tnef.js:1553:17
	tnef_pack_handle_appt_data chrome://lookout/content/tnef.js:1639:17
	tnef_pack_parse_stream chrome://lookout/content/tnef.js:1870:6
	onDataAvailable chrome://lookout/content/lookout.js:310:19
```
